src/main/python/salento/reports/map_computation/anomaly_score.py

```diff
@@ -30,18 +30,63 @@
 import json
 import argparse
 import gc
+import heapq
 # project imports
 import metric
 import data_parser
 
 
+class FilterTraces(object):
+    """ Implement different filtering options"""
+
+    @staticmethod
+    def filter_using_location(data):
+        """
+        :param data: input data list of dict, with the following keys
+                ['Index List', // index of the location at which the anomaly is lowest
+                 'Location',   // location list
+                 'Calls',      // call/event list
+                 'key',        // unique key to identify the vec
+                 'Probability', // prob vector associated with the events
+                 'Anomaly Score' // Anomaly Score
+                 ]
+        :return: filtered list of data points
+        """
+
+        unit_dict = {}
+        for value in data:
+            # group by base key
+            key = value["key"]
+            unit_key = key.split('--')[0]
+            if unit_key not in unit_dict:
+                unit_dict[unit_key] = {}
+
+            anomaly = value["Anomaly Score"]
+            for index in value["Index List"]:
+                location = value["Location"][index]
+                if location not in unit_dict[unit_key]:
+                    unit_dict[unit_key][location] = set([])
+                # add the anomaly to the location
+                unit_dict[unit_key][location].add((anomaly, key))
+
+        valid_keys = set()
+        # filter to get max anomaly at a location
+        for unit_key, value in unit_dict.items():
+            for location in value:
+                # tuple of anomaly score and key
+                max_anomaly = max(value[location])
```

Comment:
Make explicit which element of the tuple the max is being performed on to enhance clarity.